_posts/2019-02-14-Viewability.md

```diff
@@ -0,0 +1,115 @@
+
+If none of the methods is usable, we are not able to measure the viewability. In total, we are able to measure the viewability in about 80% of all delivered impressions.
+
+
+### Delivering
+
+So now we have for each impression the information if a) we could measure the viewability and b) if it was viewable.
+In general, we count unmeasurable impressions as not viewable. This is cool so far. It is very similar to a service you can buy from the IAS for instance.
+In addition, we use this information to deliver campaigns with viewable impressions as primary KPI. Thus, we want to be able to buy viewable impressions for a given price.
+The naive approach would be to take the average viewability rate of all websites and apps and decrease the bidding price by that. It is possible, that this is working, but since you would decrease the buying price by an considerable fraction, the available inventory might just be too low or too bad in quality.
+Alternatively, you could select websites and apps with a high viewability rate and only deliver on them. This approach might be feasable, but it involves a very high degree of manual optimization and you'd still need to decrease the bidding price. Plus, it only makes use of one information (the publisher) while a lot of other information might be related to the viewability rate.
+Therefore, we use machine learning as approach. The general idea is to weight each individual bid by the probability of viewability.
+
+#### Training
+
+We train two independent machine learning models, one for the measurability and one for the viewability. The measurement is a mainly technical problem and therefore is influenced by the browser version, the publisher, or the operating system.
+The viewability is influenced mostly by the publisher, the adunit itself, and the position on the site. Since the relation between those features are very different for the different metrics, we separate them in independent models.
+Based on historic data, we have a ground truth dataset in the form:
+> impression ID | measurable | viewable | features
+> 1234 | 1 | 0 | *list of features*
+> 1235 | 1 | 1 | *list of features*
+
+The features consist of various properties of the context (e.g. app/site, position on the site), the content (e.g. size of the adunit, and the user and device (e.g. operating system, browser version)
+For the measurement model we take all impressions and use the measurable column as target in a supervised classification model (namely a Neural Network).
+The output of this model is the probability of each impression to be measurable. The viewability model takes all impressions, which are measurable and uses the viewability column as target.
```

Comment:
suggest "probability of" -> "probability for"

Reply:
The training of the viewability model only uses impressions, which are measurable, and uses the viewability column as target.